### DISCUSSION TITLE
When are Requests, if ever, cloned by Hyper?

### DISCUSSION THREAD

**Discussion by Comattin** [hyperium/hyper#3741] [Q&A] [answered]
Are [Requests](https://docs.rs/hyper/latest/hyper/struct.Request.html) ever [cloned](https://docs.rs/http/1.1.0/src/http/request.rs.html#157) by Hyper. For example, are requests ever cloned part way through streaming the body? I couldn't find any instances of Request cloning when [I scanned](https://github.com/search?q=repo%3Ahyperium%2Fhyper%20.clone()&type=code) the library.

**Comment by seanmonstar** [comment, marked as answer]
No. hyper never places Clone bounds on any of the body types.
